server/gfmd/db_ldap.c

```diff
@@ -3264,6 +3264,162 @@ gfarm_ldap_quota_dir_load(void *closure,
 	return (GFARM_ERR_FUNCTION_NOT_IMPLEMENTED);
 }
 
+/**********************************************************************/
+
+static char *gfarm_ldap_user_auth_make_dn(void *vkey);
+static gfarm_error_t gfarm_ldap_user_auth_set_field(void *info,
+	char *attribute, char **vals);
+
+struct gfarm_ldap_user_auth_key {
+	const char *username;
+	const char *auth_method;
+};
+
+static const struct gfarm_ldap_generic_info_ops gfarm_ldap_user_auth_ops = {
+	&gfarm_base_user_auth_ops,
+	"(objectclass=GFarmUserAuth)",
+	"username=%s, auth_method=%s, %s",
```

Comment:
"auth_method" should be changed to "authMethod",
this name should match with the definition in gftool/config-gfarm/gfarm.schema.

also, please add the following lines to just after the fsngroupname definition in gftool/config-gfarm/gfarm.schema.
```
attributetype ( 1.3.6.1.4.1.13122.2.1.2924.101.2.32 NAME 'authMethod'
        EQUALITY caseExactMatch
        SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 SINGLE-VALUE )

attributetype ( 1.3.6.1.4.1.13122.2.1.2924.101.2.33 NAME 'authUserID'
        EQUALITY caseExactMatch
        SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 SINGLE-VALUE )
```

it is unnecessary to move the location of the GfarmUserAuth definition in gftool/config-gfarm/gfarm.schema,
because the order follows the ascending order of objectclass ID numbers.